module/VuFind/src/VuFind/RateLimiter/RateLimiterManagerFactory.php

```diff
@@ -115,6 +115,15 @@ protected function getRateLimiter(
         ?string $userId
     ): LimiterInterface {
         $policy = $config['Policies'][$policyId] ?? [];
+
+        // Truncate IP if configured, to share a quota among related IPs.
+        $ipv4Octets = $policy['groupByIpv4Octets'] ?? null;
+        $ipv6Hextets = $policy['groupByIpv6Hextets'] ?? null;
+        if ($ipv4Octets || $ipv6Hextets) {
+            $ipUtils = new \VuFind\Net\IpAddressUtils();
```

Comment:
I'd like to just reuse the instance obtained in the factory above, but not sure how to do that without passing it as a parameter into getRateLimiter, which I don't love.  Hoping there is a better option.

Reply:
You should be able to use `$this->serviceLocator->get(\VuFind\Net\IpAddressUtils::class)` here, if I'm reading things correctly.

Reply:
D'oh, I missed that variable assignment in invoke!  Fixed.